### DISCUSSION TITLE
Javascript bindings for Sveltekit like (Vite) packaging technologies

### DISCUSSION THREAD

**Discussion by netfl0** [oxigraph/oxigraph#613] [Ideas]
We are interested in using this on the D3FEND knowledge graph project to introduce people to SPARQL and semantic technology.

We currently use [sveltekit](https://kit.svelte.dev/) to create the d3fend.mitre.org website. This uses [vite](https://vitejs.dev/) as the "build" tool.

The current examples indicate for oxigraph (js)  usage with webpack is possible. Has there been any thought toward supporting svelte/vite like frameworks?

**Comment by Tpt** [comment]
Hi! Thank you for your interest in Oxigraph.

There is no built-in support for vite right now but it can be made to work. I have seen a project doing it.

A possible option I have not tested is to use https://github.com/nshen/vite-plugin-wasm-pack/ The oxigraph package distributed on node.js contains two wasm files compiled for Node and web so I hope it should be doable.

Sorry for not being more hopeful. Sadly there is no standardized way to expose wasm packages to vite at the moment.

**Reply by netfl0** [reply]
Thank you for your thoughtful response. I am just not deep on these packaging frameworks.

It looks like vite has some wasm capability, I am going to give this a try:

https://vitejs.dev/guide/features.html#webassembly

**Comment by netfl0** [comment]
this seems to be working in the latest versions of sveltekit
